GeneralsMD/Code/GameEngineDevice/Source/W3DDevice/GameClient/W3DView.cpp

```diff
@@ -226,10 +226,48 @@ void W3DView::setWidth(Int width)
  	m_3DCamera->Get_Viewport(vMin,vMax);
  	vMax.X=(Real)(m_originX+width)/(Real)TheDisplay->getWidth();
  	m_3DCamera->Set_Viewport(vMin,vMax);
+}
+
+// TheSuperHackers @tweak valeronm 25/03/2025 Adjusting FOV to look similar to what we have in in 4:3 (hFOV: 50, vFOV 38.55)
+void W3DView::adjustFovToAspectRatio(const Int width, const Int height)
+{
+	static const Real fixedHFOV = 50.00f;
+	static const Real fixedVFOV = 38.55f;
 
-	//we want to maintain the same scale, so we'll need to adjust the fov.
-	//default W3D fov for full-screen is 50 degrees.
-	m_3DCamera->Set_View_Plane((Real)width/(Real)TheDisplay->getWidth()*DEG_TO_RADF(50.0f),-1);
+	const Real aspectRatio = static_cast<Real>(width)/ static_cast<Real>(height);
+	Real vFOVRad, hFOVRad;
+	Bool drawEntireTerrain = false;
+	if (aspectRatio >= 4.0f / 3.0f)
+	{
+		// Wider than 4:3 resolutions get fixed vFOV 38.55
+		vFOVRad = DEG_TO_RADF(fixedVFOV);
+		hFOVRad = 2.0f * atan(tan(vFOVRad / 2.0f) * aspectRatio);
+		drawEntireTerrain = true;
+	}
+	else if (aspectRatio >= 1.0)
+	{
+		// Narrower than 4:3 but not yet portrait resolutions get fixed hFOV 50.0
+		hFOVRad = DEG_TO_RADF(fixedHFOV);
+		vFOVRad = 2.0f * atan(tan(hFOVRad / 2.0f) / aspectRatio);
```

Comment:
Is there no cut in fov'ness between the different conditions? How does this work?